src/Compilers/Core/Portable/Operations/IOperationEnumerable.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections;
+using System;
+using System.Diagnostics;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Implements a struct-based enumerable for <see cref="Operation"/> nodes, using a slot-based system that tracks
+    /// the current slot, and the current index in the slot if the current slot is an immutable array. This type is not hardened
+    /// to <code>default(Enumerable)</code>, and will null reference in these cases.
+    /// </summary>
+    [NonDefaultable]
+    public readonly struct IOperationEnumerable : IEnumerable<IOperation>
```

Comment:
calling this IOperationEumerable is... wonky to say the least.  Having the `I` pattern for a non-interface isn't something i've ever seen before.  I also don't think we do this with any of our other public enumerators.  I think we should follow this naming pattern:

```c#
ChildSyntaxList ChildNodesAndTokens()
```

So in this case, it would be `ChildOperationList ChildOperations`.

Reply:
Renamed.